### ISSUE TITLE
Shortcuts and Automation

### ISSUE THREAD

**Issue by EternalDeiwos** [jellyfin/Swiftfin#867]
I would like to integrate Swiftfin with the various features Apple provides for automation. Primarily this would be done by implementing actions for use in Apple Shortcuts and Home Automations.

It would be great to do things like:

- Action: Search collections for media based on title and metadata.
- Action: Watch something
- Trigger(s): Started or finished watching something
- Action(s): Add or remove favorites
- Action(s): Mark something as watched or unwatched
- ???

Given the state of the application at the time of writing I expect this will go straight to `low-priority` for now, I just wanted to get this issue recorded to create a place for discussion and generally a place to put a 👍 for those who are interested in something similar.

**Comment by linedpaper** [comment]
Agreed, would be a nice to have, but also agree there are other features that probably warrant some effort first.

**Comment by Prolonz** [comment]
Hey Team, Thanks for much for all the great work! I'm getting quite excited about the next tvOS release.

Now that the admin dashboard is implemented shortcuts could be even more useful. For example, it would be awsome to create a shortcut to run a _Scan Media Library_ task. That's particularly useful so newly added content can become available to Jellyfin immediately.

**Comment by D3478** [comment]
> Hey Team, Thanks for much for all the great work! I'm getting quite excited about the next tvOS release.
> 
> Now that the admin dashboard is implemented shortcuts could be even more useful. For example, it would be awsome to create a shortcut to run a _Scan Media Library_ task. That's particularly useful so newly added content can become available to Jellyfin immediately.


This will be lit. +999

**Comment by gd-harco** [comment]
If I got it right, App Intents got a major overhaul this year with iOS 27 (App Intents 2.0), and it's now the way the new Siri discovers and invokes app actions. SiriKit is being phased out in favor of it.
I think it could be a good step to implement this, so we could resume a specific movie/show using Siri, or search for content through Spotlight on iOS.

As a reference, the Home Assistant iOS app already implements App Intents alongside their legacy SiriKit handlers (see Sources/Extensions/AppIntents/ in their repo) for things like widget actions — it could be a useful example of how to structure this kind of dual/modern integration in Swiftfin.

(Comment translated and "enhanced" with AI as a non-english speaker, Apologies.)
